### PR TITLE
Add base model for exports with contacts, use with ticket, results and message exports

### PR DESCRIPTION
**Pull request by rowanseymour**
Provides the following consistent columns to these exports

- `Contact UUID`
- `Contact Name`
- `URN Scheme`
- `URN Value` or `Anon Value`

Next step is letting this base model add custom fields

### CODE REVIEW

File: temba/tickets/models.py [hunk 1]
```diff
@@ -652,102 +652,33 @@ def write_export(self):
...
-        exporter = TableExporter(self, "Tickets", [f["label"] for f in fields])
+        exporter = TableExporter(self, "Tickets", headers)
 
         # add tickets to the export in batches of 1k to limit memory usage
         for ticket_chunk_ids in chunk_list(ticket_ids, 1000):
             tickets = (
                 Ticket.objects.filter(id__in=ticket_chunk_ids)
                 .order_by("opened_on")
-                .prefetch_related("org")
+                .prefetch_related("org", "contact", "contact__org", "assignee", "topic")
```
Comment (rowanseymour) [inline temba/tickets/models.py:662]: @susanm74 we were missing some prefetches here - always good to have at least one `assertNumQueries` in tests

File: temba/flows/models.py [hunk 2]
```diff
@@ -1668,7 +1668,7 @@ class Meta:
         index_together = ("flow", "exit_type")
 
 
-class ExportFlowResultsTask(BaseDateRangeExport):
+class ExportFlowResultsTask(BaseWithContactExport):
```
Comment (susanm74) [inline temba/flows/models.py:1671]: just a suggestion that BaseExportWithContact could be a bit more intuitive/readable, and is also similar to BaseExportAssetStore

File: temba/flows/models.py [hunk 3]
```diff
@@ -1717,18 +1717,11 @@ def _get_runs_columns(self, extra_urn_columns, groups, contact_fields, result_fi
...
-            columns.append("URN")
+        columns += self._get_contact_headers()
 
         for extra_urn in extra_urn_columns:
             columns.append(extra_urn["label"])
 
-        columns.append("Name")
-
         for gr in groups:
             columns.append("Group:%s" % gr.name)
```
Comment (susanm74) [inline temba/flows/models.py:1726]: would an f-string be appropriate here? or are there situations where the % syntax is preferred?

File: temba/flows/tests.py [hunk 4]
```diff
@@ -4551,9 +4562,9 @@ def test_anon_org(self):
                 1,
                 [
                     self.contact.uuid,
-                    f"{self.contact.id:010d}",
-                    "tel",
                     "Eric",
+                    "tel",
+                    f"{self.contact.id:010d}",
```
Comment (susanm74) [inline temba/flows/tests.py:4567]: @rowanseymour just want to confirm this is how we generate anon values? if so, could be nice to abstract this for other exports!

**Review by susanm74 — APPROVED** [review]
just a few suggestions, otherwise looks good!

File: temba/flows/models.py (hunk 3 above)
Comment (norkans7) [reply temba/flows/models.py:1726]: f-string are preferred, it is just that that part was probably not touched since we started using that

**Review by norkans7 — APPROVED** [review]

File: temba/flows/models.py (hunk 2 above)
Comment (rowanseymour) [reply temba/flows/models.py:1671]: I think I generally prefer the last noun to be the thing that it is.. it is an export.. `BaseExportAssetStore` is an asset store, but I agree `BaseWithContact` isn't the most amazing use of ther English language... maybe `BaseItemWithContactExport`

Comment (rowanseymour) [reply temba/flows/models.py:1671]: Oh and I loath the naming of `ExportXXXTask` but that's a pain to change now

File: temba/flows/models.py (hunk 3 above)
Comment (rowanseymour) [reply temba/flows/models.py:1726]: yeah we prefer f-strings (not always possible to use them) but we haven't embarked on a repo-wide replacement, we're just updating lines as you hit them

File: temba/flows/tests.py (hunk 4 above)
Comment (rowanseymour) [reply temba/flows/tests.py:4567]: we should just call `contact.anon_display`, I copied this code and didn't change it

**Pull request merged**